stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,259 @@
+            nodes += f'transform = Transform( {scale}, 0, 0, 0, {scale}, 0, 0, 0, {scale}, 0, 0, 0 )\n'
+        nodes += "\n" # add extra new line to prepare for next node
+
+
+        ##################################
+        # iterate through all .stl files #
+        ##################################
+        meshes = data["meshes"]
+
+        # keep track of textures so we don't generate them more than once
+        texture_index = dict()
+
+        for m in meshes:
+            ####################
+            # generate texture #
+            ####################
+            texture = m["texture"]
+
+            # don't generate textures more than once
+            if texture not in texture_index:
+                ext_res_addon, ext_resource_id, sub_res_addon = generate_texture(texture, ext_resource_id, sub_resource_id)
+                ext_resource += ext_res_addon
+                sub_resource += f"{sub_res_addon}\n"
+                    
+                # add the sub resource's index to texture_index so we can use it later 
+                texture_index[texture] = sub_resource_id
+                sub_resource_id += 1 
+
+
+            #####################
+            # generate obj mesh #
+            #####################
+            stl_file = m["stl_file"]
+            stl_file_extless, _ = os.path.splitext(stl_file)
+            stl_file_location = f"{input_folder}/{stl_file}"
+
+            # output filename without a file extension
+            output_prefix = output_folder + "/" + stl_file_extless
+            
+            # Generate .obj file
+            if "limited_dissolve" in m:
+                generate_obj_file(stl_file_location, output_prefix, m["uv_map"], limited_dissolve=False)
+            else:
+                generate_obj_file(stl_file_location, output_prefix, m["uv_map"])
+
+            # add it to ext_resource
+            ext_resource += f'[ext_resource path="res://models/{output_prefix}.obj" type="ArrayMesh" id={ext_resource_id}]\n'
+            # add it to nodes
+            nodes += f'[node name="{stl_file_extless}" type="MeshInstance" parent="."]\nmesh = ExtResource( {ext_resource_id} )\nmaterial/0 = SubResource( {texture_index[texture]} )\n\n'
+
+            # move on to next ext_resource
+            ext_resource_id = ext_resource_id + 1
+
+            #######################
+            # Generate collisions #
+            #######################
+            if m["collisions"]:
+                generate_collisons(stl_file_location, output_prefix)
+                # add it to ext_resource
+                ext_resource += f'[ext_resource path="res://models/{output_prefix}.glb" type="PackedScene" id={ext_resource_id}]\n'
+                            
+                nodes += f'[node name="{stl_file_extless}Col" parent="." instance=ExtResource( {ext_resource_id} )]\n\n'
+
+                # move on to next ext_resource
+                ext_resource_id = ext_resource_id + 1
```

Comment:
if `generate_collisions` was a member function you could to all of this in that function?

Also - `generate_collisions` has return values that you don't use?